### PR TITLE
[CINN] Strong constraint branch adapt to pir

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
Others

### PR changes
Others

### Description

pcard-76996

[CINN] Strong constraint branch adapt to pir

强约束分支适配新 ir，相关数据结构、运行流程切换到新 ir。

The strong constraint branch has been modified to fit the new IR, including the corresponding data structures and running processes.

### CODE REVIEW

File: paddle/cinn/adt/generate_map_expr.cc [hunk 1]
```diff
@@ -201,6 +191,10 @@ std::shared_ptr<IGroup> MakeIGroup(const AnchorGroup& igroup_spec) {
   std::shared_ptr<const EquationFunctionConstantsProvider> constants_provider{
       new NaiveEquationFunctionConstantsProvider{
           igroup_spec.op_stmts, igroup_spec.EquationCtx4OpStmt}};
+  VLOG(1) << "IGroup spec.size() = " << igroup_spec.op_stmts->size();
```
Comment (Aurelius84) [inline paddle/cinn/adt/generate_map_expr.cc]:
```suggestion
  VLOG(4) << "IGroup spec.size() = " << igroup_spec.op_stmts->size();
```
VLOG的级别建议为4~6

File: paddle/cinn/adt/generate_map_expr.cc [hunk 2]
```diff
@@ -201,6 +191,10 @@ std::shared_ptr<IGroup> MakeIGroup(const AnchorGroup& igroup_spec) {
...
+  for (std::size_t i = 0; i < igroup_spec.op_stmts->size(); ++i) {
```
Comment (Aurelius84) [inline paddle/cinn/adt/generate_map_expr.cc]: 这里的DEBUG LOG 可以用 if(VLOG_IS_ON(4)) 包括下

File: paddle/cinn/hlir/framework/pir/op_lowering_impl.cc [hunk 3]
```diff
@@ -182,6 +186,91 @@ bool OpLowererImpl::NonFusibleScheduleDetermineFunction(::pir::Operation* op) {
...
+    std::vector<ir::Tensor>* group_func_arg_tensors,
+    std::unordered_map<::pir::Value, ir::Tensor>* tensor_map) {
+  if (FLAGS_cinn_enable_map_expr && FLAGS_cinn_enable_map_expr_schedule) {
+    apply_op_schedule = false;
+    apply_group_schedule = false;
+  }
+  VLOG(1) << "FLAGS_cinn_enable_map_expr_schedule = "
+          << FLAGS_cinn_enable_map_expr_schedule;
+  VLOG(1) << "apply_op_schedule = " << apply_op_schedule;
+  VLOG(1) << "apply_group_schedule = " << apply_group_schedule;
```
Comment (Aurelius84) [inline paddle/cinn/hlir/framework/pir/op_lowering_impl.cc]: 同上

File: paddle/phi/kernels/cpu/embedding_kernel.cc [hunk 4]
```diff
@@ -71,9 +71,9 @@ struct EmbeddingCPUFunctor {
       }
     }
 
-#if defined(_OPENMP) && !defined(PADDLE_WITH_CUDA)
-#pragma omp parallel for
-#endif
+    // #if defined(_OPENMP) && !defined(PADDLE_WITH_CUDA)
+    // #pragma omp parallel for
```
Comment (Aurelius84) [inline paddle/phi/kernels/cpu/embedding_kernel.cc]: 为什么注释掉这个？

File: paddle/cinn/adt/generate_map_expr.cc (hunk 1 above)
Comment (jiahy0825) [reply paddle/cinn/adt/generate_map_expr.cc]: Done，thanks～此处日志可直接删掉，已修改

File: paddle/cinn/adt/generate_map_expr.cc (hunk 2 above)
Comment (jiahy0825) [reply paddle/cinn/adt/generate_map_expr.cc]: 同上，thanks～

File: paddle/cinn/hlir/framework/pir/op_lowering_impl.cc (hunk 3 above)
Comment (jiahy0825) [reply paddle/cinn/hlir/framework/pir/op_lowering_impl.cc]: Done，thanks～

File: paddle/phi/kernels/cpu/embedding_kernel.cc (hunk 4 above)
Comment (jiahy0825) [reply paddle/phi/kernels/cpu/embedding_kernel.cc]: 此处误注释了，现在改回来

**Review by Aurelius84 — APPROVED** [review]

**Review by zhhsplendid — APPROVED** [review]

**Review by zhhsplendid — COMMENTED** [review]
approve for backend integration

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**